cpp/dolfinx/function/Function.cpp

```diff
@@ -38,20 +38,20 @@ la::PETScVector create_vector(const function::FunctionSpace& V)
 
   // Get dof map
   assert(V.dofmap());
-  const fem::DofMap& dofmap = *(V.dofmap());
+  std::shared_ptr<const fem::DofMap> dofmap = V.dofmap();
 
```

Comment:
Add `assert(dofmap);`

Reply:
doesn't the line above already do?

Reply:
Sorry, started to read at the bottom for some reason...